File: DAY4/원정/키로거_ 5397.py
result = []
for _ in range(int(input())):
    word = list(input().strip())
    left = []
    right = []
    for i in word:
        if i == '<':
            if left:
                right.append(left.pop())
        elif i == '>':
            if right:
                left.append(right.pop())
        elif i == '-':
            if left:
                left.pop()
        else:
            left.append(i)
    left.extend(reversed(right))
    result.append(left)

for i in result:
    print(''.join(i))     

File: DAY4/성욱/키로거 - 5397.js
const input = require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split('\n')

input.splice(0, 1)

for (const str of input) {
  const left = []
  const right = []

  for (let i = 0; i < str.length; i++) {
    if (str[i] === '<') {
      if (left.length > 0) {
        right.push(left.pop())
      }
    } else if (str[i] === '>') {
      if (right.length > 0) {
        left.push(right.pop())
      }
    } else if (str[i] === '-') {
      if (left.length > 0) {
        left.pop()
      }
    } else {
      left.push(str[i])
    }
  }
  const password = left.join('') + right.reverse().join('')
  console.log(password)
}
